server/controllers/notification.js

```diff
@@ -0,0 +1,176 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { notifyType, title, description, receivers },
+  } = req;
+
+  const sender = await Profile.findOne({ userId: user.id });
+
+  if (!sender) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const notification = await new Notification({
+    sender: sender.id,
+    notifyType,
+    title,
+    description,
+    receivers,
+  }).save();
+
+  if (notification) {
+    res.status(201).json({
+      success: {
+        notification: {
+          _id: notification.id,
+          notifyType: notification.notifyType,
+          title: notification.title,
+          description: notification.description,
+          sender: {
+            userId: user.id,
+            name: sender.name,
+            photo: sender.photo,
+          },
+          receivers: notification.receivers,
+          readBy: notification.readBy,
+          createdAt: notification.createdAt,
+        },
+      },
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid notification data");
+  }
+});
+
+exports.markNotificationRead = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { id },
```

Comment:
It's more appropriate to use [`route params`](https://expressjs.com/en/guide/routing.html#route-parameters) for this in REST.